### ISSUE TITLE
Hello World fails

### ISSUE THREAD

**Issue by rdebath** [gberger/brainfuck-in-ruby#2]
This variant of the 'Hello World' program doesn't run correctly.

```
>++++++++[<+++++++++>-]<.>>+>+>++>[-]+<[>[->+<<++++>]<<]>.+++++++..+++.>
>+++++++.<<<[[-]<[-]>]<+++++++++++++++.>>.+++.------.--------.>>+.>++++.
```

**Comment by gberger** [comment]
That's because I implemented `[` incorrectly.

The specification says:
- `[` if the byte at the data pointer is zero, then instead of moving the instruction pointer forward to the next command, jump it forward to the command after the matching ] command.
- `]` if the byte at the data pointer is nonzero, then instead of moving the instruction pointer forward to the next command, jump it back to the command after the matching [ command.

What I implemented:
- `[` noop
- `]` same as spec

This fails when the program reaches a `[` from outside when having a 0 value.
